src/sinks/postgres/integration_tests.rs

```diff
@@ -0,0 +1,305 @@
+use crate::{
+    config::{SinkConfig, SinkContext},
+    sinks::{postgres::PostgresConfig, util::test::load_sink},
+    test_util::{
+        components::{
+            run_and_assert_sink_compliance, run_and_assert_sink_error, COMPONENT_ERROR_TAGS,
+        },
+        temp_table, trace_init,
+    },
+};
+use chrono::{DateTime, Utc};
+use futures::stream;
+use serde::{Deserialize, Serialize};
+use sqlx::{Connection, FromRow, PgConnection};
+use std::future::ready;
+use vector_lib::event::{BatchNotifier, BatchStatus, BatchStatusReceiver, Event, LogEvent};
+
+const POSTGRES_SINK_TAGS: [&str; 2] = ["endpoint", "protocol"];
+
+fn pg_url() -> String {
```

Comment:
Simplified this a little bit. 

I see that https://github.com/vectordotdev/vector/pull/21248#discussion_r1801866517 stated that this should be moved to `test_utils.rs`, but as this is simplified, I don't know if its worth. 

And please not that now `PG_URL` env var is required. Before, it was optional and fallback to localhost if not set.

Take a look to the `postgresql_metrics.rs` diff 
![image](https://github.com/user-attachments/assets/720b3983-734f-4865-8f06-0478ea97acaa)

I prefer it to be simplified and not do any type of magic fallback... But I'm open to change it to what you suggest of course.